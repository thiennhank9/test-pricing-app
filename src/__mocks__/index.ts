import thunk from "redux-thunk";
import configureMockStore from "redux-mock-store";
import axiosMockAdapter from "axios-mock-adapter";
import api from "api";

const mockApi = new axiosMockAdapter(api);
const middleware = [thunk];
const mockStore = configureMockStore(middleware);

export { mockApi, mockStore };
