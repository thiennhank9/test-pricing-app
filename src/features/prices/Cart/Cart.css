.row-cart {
  display: flex;
  justify-content: flex-end;
  align-items: center;
}

.amount {
  min-width: 50px;
  display: inline-block;
  padding: 0 15px;
}
