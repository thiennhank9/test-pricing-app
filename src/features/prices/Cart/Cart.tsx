import React from "react";
import { useSelector } from "react-redux";
import { RootState } from "app/store";
import Button from "components/Button";
import { formatCurrency } from "utils";
import "./Cart.css";

function Cart() {
  const { price } = useSelector((state: RootState) => state.prices);

  return (
    <div className="row-cart">
      Order price:
      <span className="amount">{formatCurrency(price)}</span>
      <Button>Cart</Button>
    </div>
  );
}

export default Cart;
