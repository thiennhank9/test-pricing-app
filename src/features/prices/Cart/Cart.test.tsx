import React from 'react';
import { Provider } from "react-redux";
import TestRenderer from "react-test-renderer";
import { mockStore } from "__mocks__";
import Cart from "./Cart";

describe("Cart", () => {
  test("render cart", () => {
    const store = mockStore({
      prices: {
        price: 1000,
      },
    });
    const testRenderer = TestRenderer.create(
      <Provider store={store}>
        <Cart />
      </Provider>
    );
    expect(testRenderer.toJSON()).toMatchSnapshot();
    const testInstance = testRenderer.root;
    const text = testInstance.findByType("span");
    expect(text.props.children).toEqual("Â¥ 1,000");
  });
});
