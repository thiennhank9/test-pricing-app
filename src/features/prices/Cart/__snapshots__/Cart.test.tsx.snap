// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart render cart 1`] = `
<div
  className="row-cart"
>
  Order price:
  <span
    className="amount"
  >
    Â¥ 1,000
  </span>
  <button>
    Cart
  </button>
</div>
`;
